src/minidump.rs

```diff
@@ -907,21 +907,26 @@ impl<'a> MinidumpMemoryList<'a> {
     }
 
     /// Return a `MinidumpMemory` containing memory at `address`, if one exists.
-    pub fn memory_at_address(&self, address: u64) -> Option<&MinidumpMemory<'a>> {
+    pub fn memory_at_address(&self, address: u64) -> Option<&MinidumpMemory<'mdmp>> {
         self.regions_by_addr
             .get(address)
             .map(|&index| &self.regions[index])
     }
 
     /// Iterate over the memory regions in the order contained in the minidump.
-    pub fn iter<'b>(&'b self) -> MemoryRegions<'a, 'b> {
+    ///
+    /// The iterator returns items of [MinidumpMemory] as `&'slf MinidumpMemory<'mdmp>`.
+    /// That is the lifetime of the item is bound to the lifetime of the iterator itself
+    /// (`'slf`), while the slice inside [MinidumpMemory] pointing at the memory itself has
+    /// the lifetime of the [Minidump] struct ('mdmp).
+    pub fn iter<'slf>(&'slf self) -> MemoryRegions<'slf, 'mdmp> {
```

Comment:
This is actually the bug fix, the lifetimes of `MemoryRegions` where swapped.  The fix is harder to see due to the rename, but it's easier to read this correctly with the new names.